ncf_train/train.py

```diff
@@ -81,58 +64,111 @@
+    ''' Drop un-necessary columns '''
+    if 'rating' in ratings.columns:
+        train_ratings = train_ratings[['user_id', 'item_id', 'rating']]
+        test_ratings = test_ratings[['user_id', 'item_id', 'rating']]
     else:
-        recall.append(0)
-
-print("The hit ratio @ 10 is {:.2f}".format(np.average(recall)))
-torch.save(model, cnvrg_workdir + "/" + output_model_file)
+        train_ratings = train_ratings[['user_id', 'item_id']]
+        test_ratings = test_ratings[['user_id', 'item_id']]
+    return train_ratings, test_ratings
+
+def get_data_attributes(ratings, train_ratings):
+    ''' Returns data attributes '''
+    train_ratings.loc[:, 'rating'] = 1
+    num_users = ratings['user_id'].max() + 1
+    num_items = ratings['item_id'].max() + 1
+    all_item_ids = ratings['item_id'].unique()
+    return train_ratings, num_users, num_items, all_item_ids
+
+def fit_model(num_users, num_items, train_ratings, all_item_ids, batch_size, epochs, num_workers):
+    ''' Calls the training function  '''
+    model = ncf.NCF(num_users, num_items, train_ratings, all_item_ids, batch_size, num_workers)
+    trainer = pl.Trainer(max_epochs=int(epochs), reload_dataloaders_every_n_epochs=True, enable_checkpointing=False, log_every_n_steps=1)
+    # recommend_whole = pd.DataFrame(columns=['user_id', 'item_id', 'score'])
+    trainer.fit(model)
+    return model, trainer
+
+def get_test_attributes(test_ratings, ratings):
+    '''
+        returns 
+            (user:item) set
+            (user: list(all items interacted by user)) dict 
+    '''
+    test_user_item_set = set(zip(test_ratings['user_id'], test_ratings['item_id']))
+    # user1_movie_pred_whole = pd.DataFrame(columns=['user_id', 'item_id', 'rating', 'score', 'error'])
+    user_interacted_items = ratings.groupby('user_id')['item_id'].apply(list).to_dict()
+    return test_user_item_set, user_interacted_items
+
+def get_recall(test_user_item_set, user_interacted_items, all_item_ids, model):
+    ''' Calculates average recall '''
+    recall = []
+    for (u, i) in test_user_item_set:
+        interacted_items = user_interacted_items[u]
+        not_interacted_items = set(all_item_ids) - set(interacted_items)
+        selected_not_interacted = list(np.random.choice(list(not_interacted_items), 99))
+        test_items = selected_not_interacted + [i]
+
+        predicted_labels = np.squeeze(model(torch.tensor([u] * 100),
+                                            torch.tensor(test_items)).detach().numpy())
+        top10_items = [test_items[i] for i in np.argsort(predicted_labels)[::-1][0:10].tolist()]
+
+        if i in top10_items:
+            recall.append(1)
+        else:
+            recall.append(0)
+    return np.average(recall)
+
+def save_pt_model(model, file_dir, output_model_file):
+    torch.save(model, file_dir + "/" + output_model_file)
+
+if __name__ == "__main__":
+
+    warnings.simplefilter(action='ignore', category=FutureWarning)
```

Comment:
I would still wrap all this code under a function which called from if __name__ == "__main__":